src/dbc.rs

```diff
@@ -330,89 +525,117 @@ mod tests {
         // Invalid mint signatures BUT signing correct message
         for _ in 0..n_wrong_signer_sigs.coerce() {
             if let Some(input) = repeating_inputs.next() {
-                let id = crate::bls_dkg_id();
-                let key_manager =
-                    SimpleKeyManager::new(SimpleSigner::from(id.clone()), genesis_key);
+                let id = crate::bls_dkg_id(&mut rng);
+                let key_manager = SimpleKeyManager::from(SimpleSigner::from(id));
                 let trans_sig_share = key_manager
-                    .sign(&reissue_share.dbc_transaction.hash())
+                    .sign(&Hash::from(reissue_share.transaction.hash()))
                     .unwrap();
-                let trans_sig = id
-                    .public_key_set
+                let trans_sig = key_manager
+                    .public_key_set()?
                     .combine_signatures(vec![trans_sig_share.threshold_crypto()])
                     .unwrap();
                 fuzzed_transaction_sigs.insert(
-                    input.spend_key(),
-                    (id.public_key_set.public_key(), trans_sig),
+                    input.key_image.into(),
+                    (key_manager.public_key_set()?.public_key(), trans_sig),
                 );
             }
         }
 
         // Valid mint signatures BUT signing wrong message
         for _ in 0..n_wrong_msg_sigs.coerce() {
             if let Some(input) = repeating_inputs.next() {
-                let wrong_msg_sig = genesis_node.key_manager.sign(&Hash([0u8; 32])).unwrap();
-                let wrong_msg_mint_sig = genesis_node
-                    .key_manager
-                    .public_key_set()
-                    .unwrap()
+                let wrong_msg_sig = mint_node.key_manager.sign(&Hash([0u8; 32])).unwrap();
+                let wrong_msg_mint_sig = mint_node
+                    .key_manager()
+                    .public_key_set()?
                     .combine_signatures(vec![wrong_msg_sig.threshold_crypto()])
                     .unwrap();
 
-                fuzzed_transaction_sigs
-                    .insert(input.spend_key(), (genesis_key, wrong_msg_mint_sig));
+                fuzzed_transaction_sigs.insert(
+                    input.key_image.into(),
+                    (
+                        mint_node.key_manager.public_key_set()?.public_key(),
+                        wrong_msg_mint_sig,
+                    ),
+                );
             }
         }
 
         // Valid mint signatures for inputs not present in the transaction
         for _ in 0..n_extra_input_sigs.coerce() {
-            fuzzed_transaction_sigs
-                .insert(rand::random::<SpendKey>(), (genesis_key, mint_sig.clone()));
+            fuzzed_transaction_sigs.insert(
+                Default::default(),
```

Comment:
Feels a bit loose to have a `Default` impl on `KeyImage`, How about we just be explicit with a `KeyImage::random(&mut rng)` and drop the Default?

Reply:
done.   added ::random(), and removed derive Default.

```
    pub fn random(rng: &mut impl rand8::RngCore) -> Self {
        Self(G1Projective::random(rng).to_affine())
    }
```